File: lib/java/src/ae/math/ReadOnlyBackend.java
package ae.math;

public final class ReadOnlyBackend extends VectorBackend {
	
	private final VectorBackend _backend;
	
	@Override
	protected final float _getElement(final int index) {
		return _backend.getElement(index);
	}
	
	@Override
	protected final float _getX() {return _backend.getX();}
	@Override
	protected final float _getY() {return _backend.getY();}
	@Override
	protected final float _getZ() {return _backend.getZ();}
	@Override
	protected final float _getW() {return _backend.getW();}

	@Override
	protected final void _setElement(
		final int   index,
		final float value) {}
	
	@Override
	protected final void _setX(final float x) {}
	@Override
	protected final void _setY(final float y) {}
	@Override
	protected final void _setZ(final float z) {}
	@Override
	protected final void _setW(final float w) {}

	public ReadOnlyBackend(
			final VectorBackend backend) {
		
		_backend = backend;
	}
}

File: lib/javascript/src/classes/math_VectorBackend/ReadOnlyBackend.js

// ae.math.ReadOnlyBackend
class AEClassReadOnlyBackend extends AEClassVectorBackend {
	
	_backend: ae.math.VectorBackend;
	
	constructor(
			backend: ae.math.VectorBackend) {
		
		super();
		this._backend = backend;
	}
	
	get x(): number {return this._backend.x;}
	get y(): number {return this._backend.y;}
	get z(): number {return this._backend.z;}
	get w(): number {return this._backend.w;}
	
	set x(x: number): ae.math.VectorBackend {return this;}
	set y(y: number): ae.math.VectorBackend {return this;}
	set z(z: number): ae.math.VectorBackend {return this;}
	set w(w: number): ae.math.VectorBackend {return this;}
	
	getElement(index: number): number {
		
		return this._backend.getElement(index);
	}
	
	setElement(
			index: number,
			value: number): ae.math.VectorBackend {
		
		return this;
	}
}
